pkg/sql/session.go

```diff
@@ -368,12 +370,12 @@ func (ts *txnState) resetStateAndTxn(state TxnStateEnum) {
 // The session context is just used for logging the SQL trace.
 func (ts *txnState) finishSQLTxn(sessionCtx context.Context) {
 	ts.mon.Stop(ts.Ctx)
-	if ts.sp == nil {
-		panic("No span in context? Was resetForNewSQLTxn() called previously?")
+	var sampledFor7881 bool
+	if ts.sp != nil {
+		sampledFor7881 = (ts.sp.BaggageItem(keyFor7881Sample) != "")
```

Comment:
For safety, right? The code feels less readable without them, though I don't particularly care either way, so I've left it alone.